acme_diags/plot/cartopy/enso_diags_plot.py

```diff
@@ -0,0 +1,268 @@
+        return 30
+    elif degrees_covered > 20:
+        return 10
+    else:
+        return 1    
+
+def plot_panel(n, fig, proj, var, clevels, cmap,
+               title, parameter, conf=None, stats={}):
+
+    var = add_cyclic(var)
+    lon = var.getLongitude()
+    lat = var.getLatitude()
+    var = ma.squeeze(var.asma())
+
+    # Contour levels
+    levels = None
+    norm = None
+    if len(clevels) > 0:
+        levels = [-1.0e8] + clevels + [1.0e8]
+        norm = colors.BoundaryNorm(boundaries=levels, ncolors=256)
+
+    # Contour plot
+    ax = fig.add_axes(panel[n], projection=proj)
+    region_str = parameter.regions[0]
+    region = regions_specs[region_str]
+    if 'domain' in region.keys():
+        # Get domain to plot
+        domain = region['domain']
+    else:
+        # Assume global domain
+        domain = cdutil.region.domain(latitude=(-90., 90, 'ccb'))
+    kargs = domain.components()[0].kargs
+    lon_west, lon_east, lat_south, lat_north = (0, 360, -90, 90)
+    if 'longitude' in kargs:
+        lon_west, lon_east, _ = kargs['longitude']
+    if 'latitude' in kargs:
+        lat_south, lat_north, _ = kargs['latitude']
+    lon_covered = lon_east - lon_west
+    lon_step = determine_tick_step(lon_covered)
+    xticks = np.arange(lon_west, lon_east, lon_step)
+    xticks = np.append(xticks, lon_east)
+    lat_covered = lat_north - lat_south
+    lat_step = determine_tick_step(lat_covered)
+    yticks = np.arange(lat_south, lat_north, lat_step)
+    yticks = np.append(yticks, lat_north)
+    ax.set_extent([lon_west, lon_east, lat_south, lat_north], crs=proj)
+    cmap = get_colormap(cmap, parameter)
+    contours = ax.contourf(lon, lat, var,
+                           transform=ccrs.PlateCarree(),
+                           norm=norm,
+                           levels=levels,
+                           cmap=cmap,
+                           extend='both'
+    )
+
+    if conf is not None:
+        conf = add_cyclic(conf)
+        conf = ma.squeeze(conf.asma())
+        # Values in conf will be either 0 or 1. Thus, there are only two levels -
+        # represented by the no-hatching and hatching levels.
+        ax.contourf(lon, lat, conf,2,
+                    transform=ccrs.PlateCarree(),
+                    norm=norm,
+                    colors='none',
+                    extend='both',
+                    hatches=[None, '//']
+        )
+    # Full world would be aspect 360/(2*180) = 1
+    ax.set_aspect((lon_east - lon_west)/(2*(lat_north - lat_south)))
+    ax.coastlines(lw=0.3)
+    if title[0] is not None:
+        ax.set_title(title[0], loc='left', fontdict=plotSideTitle)
+    if title[1] is not None:
+        ax.set_title(title[1], fontdict=plotTitle)
+    if title[2] is not None:
+        ax.set_title(title[2], loc='right', fontdict=plotSideTitle)
+    #ax.set_xticks(xticks, crs=ccrs.PlateCarree())
+    ax.set_xticks([0, 60, 120, 180, 240, 300, 360.01], crs=ccrs.PlateCarree())
```

Comment:
We should probably try to  fix how `xticks` is generated above rather than hard-coding the `xticks` here.

Reply:
The first tick label show both 0E and 0W with the calculated `xticks`